src/exprs.c

```diff
@@ -1053,18 +1065,25 @@ void ListExpr2 (
           {
             if (posshole == 1)
               {
-                SET_FILT_LIST(list, FN_IS_NDENSE);
+                if (!safe || TNUM_OBJ(list) == T_PLIST) {
```

Comment:
Hmm, is that safe? Couldn't the tnum be, say `T_PLIST_DENSE` at this point? That would suggest using `if (!safe || IS_PLIST(list))` - or perhaps even `IS_LIST`?

But why is this necessary at all? `SET_FILT_LIST` uses a tnum dispatch table -- if the user did something really bad here, wouldn't that just trigger an error via that dispatch table?

Reply:
You are right, let me rethink this.